src/rubrix/client/datasets.py

```diff
@@ -841,6 +894,40 @@ def spans2iob(example):
 
         return ds.cast(new_features)
 
+    @_requires_spacy
+    def _prepare_for_training_with_spacy(
+        self, nlp: "spacy.Language"
+    ) -> "spacy.tokens.DocBin":
+
+        from spacy.tokens import DocBin
+
+        db = DocBin()
+
+        # Creating the DocBin object as in https://spacy.io/usage/training#training-data
+        for record in self._records:
+            if record.annotation is None:
+                continue
+
+            doc = nlp(record.text)
+            entities = []
+
+            for anno in record.annotation:
+                span = doc.char_span(anno[1], anno[2], label=anno[0])
+                # There is a misalignment between record tokenization and spaCy tokenization
+                if span is None:
+                    # TODO(@dcfidalgo): Do we want to warn and continue or should we stop the training set generation?
```

Comment:
I would raise an exception here since, in this case, the user should likely revise the tokenization (the spacy language model provided). Maybe in a second iteration, we could think about adding an argument that turns this exception into a warning